lib/registrant.rb

```diff
@@ -0,0 +1,53 @@
+class Registrant
+  attr_reader :age, :name, :permit, :license_data
+
+  def initialize(name, age, permit = false)
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = {written: false, license: false, renewed: false}
+  end
+
+  def permit?
+    permit
+  end
+
+  def written?
+    license_data[:written]
+  end
+
+  def license?
+    license_data[:license]
+  end
+
+  def earn_permit
+    @permit = true
+  end
+
+  def take_written
```

Comment:
Remember that methods that return booleans should end in a `?`

Reply:
And maybe this method name would be more descriptive as `can_take_written?` since it's doing a check for that.